### PR TITLE
Add SPV_EXT_arithmetic_fence extension

### PR DESCRIPTION
**Pull request by MrSidims**

### CODE REVIEW

File: extensions/INTEL/SPV_INTEL_arithmetic_fence.asciidoc [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+can be transformed into:
+
+[source,C]
+OpArithmeticFenceINTEL(a + (b + c)) + d
+
+but not into:
+
+[source,C]
+OpArithmeticFenceINTEL(a + b) + (c + d)
```
Comment (kpet) [inline extensions/INTEL/SPV_INTEL_arithmetic_fence.asciidoc:75]: These examples are useful but would IMO be more readable if fully expressed as pseudo SPIR-V assembly (omitting result types):
```
%ab = OpFAdd %a %b
%abc = OpFAdd %ab %c
%abc_fence = OpArithmeticFenceINTEL %abc
%result = OpFAdd %abc_fence %d
```

can be transformed into:

```
%bc = OpFAdd %b %c
%abc = OpFAdd %a %bc
%abc_fence = OpArithmeticFenceINTEL %abc
%result = OpFAdd %abc_fence %d
```

but not into:

```
%ab = OpFAdd %a %b
%ab_fence = OpArithmeticFenceINTEL %ab
%cd = OpFAdd %c %d
%result = OpFAdd %ab_fence %cd
```

**Review by kpet — COMMENTED** [review]
We would be interested in seeing this published as an EXT instead.

**Comment by MrSidims** [comment]
I don't mind to change it from vendor extension to EXT. Converting to draft for now

File: extensions/INTEL/SPV_INTEL_arithmetic_fence.asciidoc (hunk 1 above)
Comment (MrSidims) [reply extensions/INTEL/SPV_INTEL_arithmetic_fence.asciidoc]: Updated, thanks!

File: extensions/EXT/SPV_EXT_arithmetic_fence.asciidoc [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+--
+
+Instructions
+~~~~~~~~~~~~
+
+In section 3.49.1. Miscellaneous Instructions, add a new instruction:
+
+[cols="5", width="100%"]
+|=====
+4+^|*OpArithmeticFenceEXT* +
```
Comment (bashbaug) [inline extensions/EXT/SPV_EXT_arithmetic_fence.asciidoc:147]: Without this change, asciidoctor centers the contents of this table cell, which probably isn't desired.

```suggestion
4+|*OpArithmeticFenceEXT* +
```

File: extensions/EXT/SPV_EXT_arithmetic_fence.asciidoc [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+Return 'Target'. Indicates that the optimizer can not move or combine 'Target'
+with the expression that uses 'Result' of the instruction. +
+ +
+'Target' must be scalar or vector of _floating-point type_. +
```
Comment (bashbaug) [inline extensions/EXT/SPV_EXT_arithmetic_fence.asciidoc:151]: This extension seems to be rendering fine, but when I was updating the Intel extensions for newer asciidoctor toolchains I found that using `'single quotes'` didn't reliably produce italic text, whereas using `_underscores_` did.  If you want to be safe, consider switching to underscores now, but it doesn't look like this is currently required.

File: README.md [hunk 4]
```diff
@@ -74,6 +74,7 @@ Khronos SPIR-V Registry](https://www.khronos.org/registry/spir-v/).
 * [SPV_EXT_shader_tile_image               ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_tile_image.html)
 * [SPV_EXT_shader_viewport_index_layer     ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_viewport_index_layer.html)
 * [SPV_EXT_ycbcr_attachments               ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_ycbcr_attachments.html)
+* [SPV_EXT_arithmetic_fence __             ]( http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_arithmetic_fence.html)
```
Comment (bashbaug) [inline README.md:77]: Could you please try to keep the list of extensions alphabetized, for easier navigation?

**Review by bashbaug — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as discussed in the August 28th teleconference.

**Pull request merged**